intents/oversights/benchmark_set_too_different.py

```diff
@@ -0,0 +1,180 @@
+
+    """ required_table_for_all is a table which has all the data which was in
+        the initial table but with the comparision column as '*' because we
+        have to compare single value with the whole table"""
+    required_table_for_all = table.copy()
+    required_table_for_all[slice_compare_column] = '*'
+
+    updated_table = pandas.concat([required_table_for_one, required_table_for_all])
+
+    grouping_columns = []
+    if dimensions is not None:
+        grouping_columns = dimensions.copy()
+    grouping_columns.append(slice_compare_column)
+    
+    result_table = aspects.group_by(updated_table, grouping_columns, summary_operator)
+
+    other_metrics = all_metric.copy()
+    other_metrics.remove(metric)
+
+    columns_order = grouping_columns.copy()
+    columns_order.append(metric)
+    columns_order = columns_order + other_metrics
+
+    """ We have done the reordering of columns so that all the dimension
+        should appear first followed by the metric given by the user and
+        and then rest of the columns."""
+    result_table = result_table[columns_order]
+
+    num_rows = result_table.shape[0]
+    num_columns = result_table.shape[1]
+    dimensions_len = 0
+    if dimensions is not None:
+        dimensions_len = len(dimensions)
+
+    result_table_matrix = result_table.values.tolist()
+
+    suggestion = []
+
+    # We have to iterate through all metric which was not involved 
+    # in the computation initially.
+    for column_i in range(dimensions_len + 2, num_columns):
+        # it can store the index of row on whcih the oversight might appear.
+        column_i_suggestion_list = []
+        row_i = 0
+        while row_i < num_rows:
+            if row_i == num_rows - 1 or result_table_matrix[row_i][:dimensions_len] != result_table_matrix[row_i + 1][:dimensions_len]:
+                column_i_suggestion_list.append(row_i)
+            else:
+                if _calculate_relation(result_table_matrix[row_i][column_i], result_table_matrix[row_i + 1][column_i]) < constants.BSTD_DISIMILARITY_THRESHOLD:
+                    row_i = row_i + 1
+                elif _calculate_relation(result_table_matrix[row_i][dimensions_len + 1], result_table_matrix[row_i + 1][dimensions_len + 1]) < constants.BSTD_DISIMILARITY_THRESHOLD:
+                    row_i = row_i + 1
+                else:
+                    column_i_suggestion_list.append(row_i)
+                    column_i_suggestion_list.append(row_i + 1)
+                    row_i = row_i + 1
+            row_i = row_i + 1
+
+        if len(column_i_suggestion_list) > 0:
+            """ for every suggestion we form a dictionary and 
+                append it to the list of dictionary."""
+            suggestion_i = {}
+            suggestion_i['suggestion'] = slice1 + ' looks different from everyone else on ' + metric + '. You might also want to look  at ' + columns_order[column_i] +' since ' + slice1 + ' also looks different on this.'
```

Comment:
The word "everyone" might not apply to all data sets. Please make this message more generic.

Reply:
this message was suggested by sendhil in one meeting so I just used it. can we write 
'A looks different with others on metric. you might also want to look at suggested_metric since A also looks different on this.'?
I shall think for other wording also and let you know once I shall come up with any suggestion.